lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -254,6 +254,27 @@ static bool printPackedTypeImpl(Type type, raw_ostream &os, Operation *op,
         op->emitError("Unexpected unpacked array in packed type ") << arrayType;
         return true;
       })
+      .Case<TypeAliasType>([&](TypeAliasType typeRef) {
+        SymbolRefAttr ref = typeRef.getRef();
+        auto moduleOp = op->getParentOfType<ModuleOp>();
+
+        auto typeScope =
+            moduleOp.lookupSymbol<TypeScopeOp>(ref.getRootReference());
+        if (!typeScope) {
+          op->emitError("unable to resolve scope for type reference");
+          return false;
+        }
+
+        auto typedecl =
+            typeScope.lookupSymbol<TypedeclOp>(ref.getLeafReference());
+        if (!typedecl) {
+          op->emitError("unable to resolve declaration for type reference");
+          return false;
+        }
+
+        os << typedecl.verilogName().getValueOr(typedecl.getName());
```

Comment:
nit: Getting the name to use should probably be a helper function on typedecl op.